### PR TITLE
[libc][math][c23] Add {nearbyint,rint,lrint,llrint,lround,llround}f16 C23 math functions

### PR DESCRIPTION
**Pull request by overmighty**
cc @lntue

### CODE REVIEW

**Comment by overmighty** [comment]
Rebased and resolved the merge conflicts.

File: libc/src/math/generic/llrintf16.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//===----------------------------------------------------------------------===//
+
+#include "src/math/llrintf16.h"
+#include "src/__support/FPUtil/NearestIntegerOperations.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(long long, llrintf16, (float16 x)) {
+  return fputil::round_to_signed_integer_using_current_rounding_mode<float128,
```
Comment (lntue) [inline libc/src/math/generic/llrintf16.cpp:16]: `float16`?

File: libc/src/math/generic/llroundf16.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//===----------------------------------------------------------------------===//
+
+#include "src/math/llroundf16.h"
+#include "src/__support/FPUtil/NearestIntegerOperations.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(long long, llroundf16, (float16 x)) {
+  return fputil::round_to_signed_integer<float128, long long>(x);
```
Comment (lntue) [inline libc/src/math/generic/llroundf16.cpp:16]: `float16`?

File: libc/src/math/generic/lroundf16.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+//===----------------------------------------------------------------------===//
+
+#include "src/math/lroundf16.h"
+#include "src/__support/FPUtil/NearestIntegerOperations.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(long, lroundf16, (float16 x)) {
+  return fputil::round_to_signed_integer<float128, long>(x);
```
Comment (lntue) [inline libc/src/math/generic/lroundf16.cpp:16]: `float16`?

**Review by lntue — APPROVED** [review]

**Pull request merged**